feat(utilities): add scale_wind_speed function - Shunhao Li

windsoul124 opened the pull request:
# Description
## Summary
This PR implements the new function `scale_wind_speed` as requested in Issue #204.
The function scales a 10 m wind speed to an arbitrary height using a logarithmic wind profile.

This implementation follows the convention used in UTCI-related literature (Bröde et al., 2012) and adapts the approach implemented in the [thermofeel](https://github.com/ecmwf/thermofeel/blob/62f754b7fe89fefc0789b68f9b96e58952386377/thermofeel/thermofeel.py#L131).

### Motivation and Context
- Ensures wind speed data collected at heights other than 10 m can be scaled consistently. 
- Required for accurate thermal comfort calculations within the UTCI framework.  

### Impact on other components
- This feature is self-contained within `utilities.py`
- No breaking changes introduced
- No modifications required in existing models or workflows

Fixes #204 
## Type of change
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?
- Added a dedicated test file: `tests/test_scale_wind_speed.py`
- Tests include:  
  - [x] Scaling with scalar inputs  
  - [x] Scaling with array inputs  
  - [x] Edge case at 10 m (should return original speed)  
  - [x] Invalid input (`h <= 0.01`) raises `ValueError`
 - Ran unit tests via:
  ```bash
  pytest tests/test_scale_wind_speed.py -v
  tox -e py312
  tox -e docs
```
- All tests and documentation builds passed successfully.
 #### Test configuration
- Python version: 3.12
- Environment: pipenv (from provided `Pipfile`)
- Platform: Windows 10 Pro 22H2
- IDE: PyCharm 2022.3.3

# Checklist  
- [x] My code follows the style guidelines of this project  
- [x] I have performed a self-review of my code  
- [x] I have commented my code, particularly in hard-to-understand areas  
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings  
- [x] I have added tests that prove my feature works  
- [x] New and existing unit tests pass locally with my changes

FedericoTartarini commented on pythermalcomfort/utilities.py:
```diff
@@ -1245,3 +1245,41 @@ class DefaultSkinTemperature(NamedTuple):
+    va: float or list of floats
+        wind speed measured at 10 m above ground level, [m/s]
+    h : float or list of floats
+        target height at which to scale the wind speed, [m]; must be > 0.01 m
```
I know this was not shown in the example code in the issue but you should also have exposed z0 as input to the function. See code from the other candidates.

FedericoTartarini commented on pythermalcomfort/utilities.py:
```diff
@@ -1245,3 +1245,41 @@ class DefaultSkinTemperature(NamedTuple):
+    va = np.array(va)
+    h = np.array(h)
+
+    if np.any(h <= 0.01):
```
great that you included this check, but you are not checking the validity of the other inputs

windsoul124 replied:
Good point - I've expanded the validation checks to cover the inputs.

FedericoTartarini commented on tests/test_scale_wind_speed.py:
```diff
@@ -0,0 +1,32 @@
+import numpy as np
+import pytest
+
+from pythermalcomfort.utilities import scale_wind_speed
```
great that you included some tests

windsoul124 replied:
Thanks. I've added more tests to mirror all validation paths:
- Non-finite inputs (`va`, `h`, `z0`)
- Negative `va`
- `z0` domain
- Non-broadcastable shapes
- scalar and vector broadcasting

FedericoTartarini reviewed (commented):
thank you for contributing to pythermalcomfort. Very nice work, I left a few comments for you to address.

You can review the other candidates code and make changes now since everyone has submitted their work.

windsoul124 commented:
@FedericoTartarini Thank you for the review and for the suggestions.
I've gone through your comments and addressed them in my latest changes. I also reviewed other candidates' submissions and incorporated improvements where appropriate.
